bundles/org.openhab.automation.jsscripting/src/main/java/org/openhab/automation/jsscripting/internal/fs/watch/JSDependencyTracker.java

```diff
@@ -0,0 +1,48 @@
+/**
+ * Copyright (c) 2010-2021 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+
+package org.openhab.automation.jsscripting.internal.fs.watch;
+
+import java.io.File;
+
+import org.openhab.core.OpenHAB;
+import org.openhab.core.automation.module.script.rulesupport.loader.DependencyTracker;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Deactivate;
+
+/**
+ * Tracks JS module dependencies
+ *
+ * @author Jonathan Gilbert - Initial contribution
+ */
+@Component(immediate = true, service = JSDependencyTracker.class)
+public class JSDependencyTracker extends DependencyTracker {
+
+    public static final String LIB_PATH = String.join(File.separator, OpenHAB.getConfigFolder(), "automation", "js",
```

Comment:
Since this directory does not exist yet, should we attempt to create it when this bundle is activated?  I think if the user creates it after the script watcher starts to monitor, they would need to restart OH (or the related bundles) to have the watcher see it.

Reply:
Hmm, good point about the restarting. I wondered if it would start watching once created, but it doesn't look like it does.

Reply:
I've added this in now, just like for the script watching.

Reply:
Awesome.